packages/simcore-sdk/src/simcore_sdk/node_ports_common/filemanager.py

```diff
@@ -309,68 +310,88 @@ async def upload_file(
     )
+    num_retries: int = (
+        NodePortsSettings.create_from_envs().NODE_PORTS_IO_NUM_RETRY_ATTEMPTS
+    )
     if io_log_redirect_cb:
         await io_log_redirect_cb(f"uploading {file_to_upload}, please wait...")
     async with ClientSessionContextManager(client_session) as session:
-        upload_links = None
-        try:
-            store_id, upload_links = await get_upload_links_from_s3(
-                user_id=user_id,
-                store_name=store_name,
-                store_id=store_id,
-                s3_object=s3_object,
-                client_session=session,
-                link_type=LinkType.S3 if use_rclone else LinkType.PRESIGNED,
-                file_size=ByteSize(
-                    file_to_upload.stat().st_size
-                    if isinstance(file_to_upload, Path)
-                    else file_to_upload.file_size
-                ),
-            )
-            # NOTE: in case of S3 upload, there are no multipart uploads, so this remains empty
-            uploaded_parts: list[UploadedPart] = []
-            if use_rclone:
-                assert r_clone_settings  # nosec
-                assert isinstance(file_to_upload, Path)  # nosec
-                await r_clone.sync_local_to_s3(
-                    file_to_upload,
-                    r_clone_settings,
-                    upload_links,
-                )
-                await progress_bar.update()
-            else:
-                uploaded_parts = await upload_file_to_presigned_links(
-                    session,
-                    upload_links,
-                    file_to_upload,
-                    num_retries=NodePortsSettings.create_from_envs().NODE_PORTS_IO_NUM_RETRY_ATTEMPTS,
-                    io_log_redirect_cb=io_log_redirect_cb,
-                    progress_bar=progress_bar,
-                )
-
-            # complete the upload
-            e_tag = await _complete_upload(
-                session,
-                upload_links,
-                uploaded_parts,
-            )
-        except (r_clone.RCloneFailedError, exceptions.S3TransferError) as exc:
-            log.error("The upload failed with an unexpected error:", exc_info=True)
-            if upload_links:
-                await _abort_upload(session, upload_links, reraise_exceptions=False)
-            raise exceptions.S3TransferError from exc
-        except CancelledError:
-            if upload_links:
-                await _abort_upload(session, upload_links, reraise_exceptions=False)
-            raise
-        if io_log_redirect_cb:
-            await io_log_redirect_cb(f"upload of {file_to_upload} complete.")
-        return store_id, e_tag
+        async for attempt in AsyncRetrying(
+            reraise=True,
+            wait=wait_exponential(min=1, max=10),
+            stop=stop_after_attempt(num_retries),
+            retry=retry_if_exception_type(CancelledError)
+            | retry_if_not_result(use_rclone),

```

Comment:
1. I also agree with @pcrespov comment above. especially since this has a tendency of error we should have well created logs and the decorator makes them much nicer to read.

2. Then I think the wait_exponential I think we use another one usually which is the wait_random_exponential one. we decided on using it some time ago. 

3. I think ```retry_if_exception_type(CancelledError)``` is not the one to choose here. We can check together which one to use, but the one of the 400 is not a cancellation. Also if someone actually cancels, then this code would retry again which would be very unfortunate. Let's check that one together.